File: client/src/app/courses/shared/course.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class CourseService {

  private baseUrl = `${environment.baseUrl}courses`;

  constructor(private http: HttpClient) { }

  createCourse(course: Object): Observable<any> {
    return this.http.post(`${this.baseUrl}/create`, course);
  }

  getCourseById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  getCoursesList(): Observable<any> {
    console.log(this.baseUrl);
    return this.http.get(`${this.baseUrl}`);
  }

  deleteCourse(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text'});
  }

  updateCourse(id: number, value: any): Observable<Object> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }
}

File: server/src/main/java/com/agilexp/service/CourseService.java
package com.agilexp.service;

import com.agilexp.dbmodel.Course;

import java.util.Collection;
import java.util.List;

public interface CourseService {

    public abstract Course createCourse(Course course);
    public abstract Course getCourseById(long id);
    public abstract List<Course> getCourses();
    public abstract boolean updateCourse(long id, Course course);
    public abstract void deleteCourse(long id);
    public abstract void deleteAllCourses();
}
